### ISSUE TITLE
Skip "Select a job title" onboarding step for sign ups from Landing Pages

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#331]
- If they select "Other" or come from the learn more without selecting a job title, then they do get step 1, /welcome.
- in addJobTitles activity, there's a getJobTitle parameter - we should modify/move this script.
- If jobTitleID, add entry into the db with the jobTitle automatically and skip the onboarding step

**Comment by situatedbit** [comment]
estimate: 3

**Comment by IagoSRL** [comment]
3

**Comment by joshdanielson** [comment]
2

**Comment by IagoSRL** [comment]
I think this must implemented by attaching the selected jobTitleID into the sign-up request, updating the REST API, and in that process the user gets added the job title immediately and returns an onboardingStep parameter that, rather than being the default value 'welcome', is the step after 'addJobTitles'.

This way we directly move the user to the correct step when the signup completes, rather than add additional logic into the 'addJobTitles' activity.

Additionally, current behavior in 'addJobTitles' to preselect a jobTitleID can be removed to simplify it, since gets replaced with this new feature.

@situatedbit @joshdanielson Let me know if you think in a different approach or any comment.

**Comment by situatedbit** [comment]
@IagoSRL I think you're approach above looks good.

**Comment by IagoSRL** [comment]
I just found that, additionally to allow the user to pick a jobTitleID, we allow them to request us to create a new job title based on a given name, if not found in the job title autocomplete.
So, in this, additionally to the new jobTitleID I will add yet another parameter in the singup for the jobTitleName provided, so behave the same as addJobTitles.

**Comment by IagoSRL** [comment]
This is implemented, ready for QA.

Some notes:
- The server uses the same code than addJobTitle for the job titles given by the sign-up, so providing a jobTitleID or a new jobTitleName has the same behavior than there.
- I didn't removed the support for passing in preset job title into addJobTitles as set in this issue, because there is additional code using that; exaclty, and edge case where a logged user goes to learnMoreProfessionals and add a job title from there (but there is no sign-up, so is just redirect to addJobTitles). I think is better to re-think if all this is wanted at another time.
- if adding the job title when processing sign-up fails, rather than throw error to the user I'm skipping and let the process to continue, leading to the user going to the first onboarding step;
  - to be clear: sign-up and add-job-title are two independent processes, just run one after the other; the sign-up could have worked perfectly while the add-job-titles not
  - is really strange the add-job-title to fail using our UI, so extra work on this seems not reasonable
  - in case that happens anyway, the user will have the opportunity to try again adding the job title at addJobTitles activity, with detailed errors there --may be not the best usability to retry but is better than throw the error at the sign-up form, where user can think the sign-up failed (but was OK, the account is already created). Still, very edge case, but just wanted to point this out.

It's on *dev*

**Comment by situatedbit** [comment]
I forgot to add siteUrl on dev and production. As a result the API call fails on dev. I've fixed this (I think), and merged it into #331. But, it's not pushed to dev yet.

**Comment by joshdanielson** [comment]
@IagoSRL I can't test this until this branch has the job title list in the form so waiting on #367 to be completed and merged into this.

**Comment by IagoSRL** [comment]
In *dev* was uploaded #367.

I have commented changes since there pending of upload, working on them though

**Comment by joshdanielson** [comment]
Okay, I know we made it confusing with so many branches! I just tested it on dev and it works...but it skips the /welcome step, too. 

Two ways I can think to resolve it easily:
1. Skip the welcome page (well leave it the way it is now). I'm not sure if that's wise or not. 
2. Take the welcome message for service professionals and make it a pop-up on /scheduling preferences. The message is short and I removed the help links (on desktop at least) so it's just a quick message.

I'd like @situatedbit 's thoughts but that won't be until tomorrow...so if it's dependent on releasing today, let's just let it skip for this round.

**Comment by joshdanielson** [comment]
@IagoSRL and either of those two options are probably better than /welcome being its own page for service professionals. Again, let's ask @situatedbit.

**Comment by IagoSRL** [comment]
@joshdanielson Please put that in a new issue. It looks small, maybe we can comment to add the welcome for clients too (there is text for clients there I think, but is never displayed to them). and we move this on.

And yes, I knew that the welcome was skipped, just what we do is to set the initial onboarding step to be the scheduling, so any previous (welcome, addJobTitle) never are show on that case. I though minor since users comes from some kind of presentation page (the landing or learnMore), but is fine to add it again as a pop-up I think, we will see.

**Comment by joshdanielson** [comment]
@IagoSRL okay! I'll chat with @situatedbit first and decide if we need a new issue.

**Comment by IagoSRL** [comment]
OK, thanks! Closing
